salt/utils/boto.py

```diff
@@ -179,8 +179,10 @@ def get_connection(service, module=None, region=None, key=None, keyid=None,
     '''
 
     module = module or service
+    module = 'boto.' + module
+    module, submodule = module.rsplit('.', 1)
```

Comment:
Sorry, as I see your code does exactly the same as mine, and your version looks more Pythonic, but it slower and less readable. Probably you're right about the triple `module` assigning. I've rewrote the last two lines to a single one.
```bash
$ python -m timeit -s "module = 'a.b.c.d'" "module, submodule = (lambda mod: ['.'.join(pk) for pk in [mod[:-1], mod[-1:]]])(['boto'] + module.split('.'))"
1000000 loops, best of 3: 1.12 usec per loop
$ python -m timeit -s "module = 'a.b.c.d'" "module = 'boto.' + module; module, submodule = module.rsplit('.', 1)"
1000000 loops, best of 3: 0.285 usec per loop
$ python -m timeit -s "module = 'a.b.c.d'" "module, submodule = ('boto.' + module).rsplit('.', 1)"
1000000 loops, best of 3: 0.281 usec per loop
```